akka-actor-typed-tests/src/test/resources/logback-doc-dev.xml

```diff
@@ -17,14 +17,20 @@
         </encoder>
     </appender>
 
-    <appender name="ASYNC" class="ch.qos.logback.classic.AsyncAppender">
+    <appender name="ASYNC-FILE" class="ch.qos.logback.classic.AsyncAppender">
         <queueSize>1024</queueSize>
         <neverBlock>true</neverBlock>
         <appender-ref ref="FILE" />
     </appender>
 
+    <appender name="ASYNC-STDOUT" class="ch.qos.logback.classic.AsyncAppender">
+        <queueSize>1024</queueSize>
+        <neverBlock>true</neverBlock>
```

Comment:
should mention this may lose log

Reply:
Would be good to clarify this ("when less than 20% of the queue capacity remains, AsyncAppender will drop events of level TRACE, DEBUG and INFO keeping only events of level WARN and ERROR" - does it log anything so you at least know log messages are missing?)

Reply:
In all the benchmarks I found, a considerable amount of messages get dropped using a similar setup. So, maybe we should provide a better default than this?

The improvement of using an async appender over a synchronous one seems to be in the order of a few µs/log message, so we are definitely in micro-optimization terrain if you care about that.

Also there seem to be [some doubts](https://jira.qos.ch/browse/LOGBACK-1279) whether logbacks AsyncAppenders scale with high throughput.

I think we should have a default which doesn't drop anything and if we want to add some hints about how to optimize if you really need more speed than that.

Reply:
These settings were reviewed by @wsargent , which I think has more real world logging experience than we have.

I mean, for the dev mode stdout it's not important to use the async appender, but for file based appenders we don't want to block the caller.

Reply:
> for file based appenders we don't want to block the caller.

If the choice is between losing messages and blocking a bit longer (appending to a queue also "blocks the caller"), the default should be not to lose logging messages. I you want to shave off a few more µs/message, you would probably go with log4j2 anyway as that is even faster anyway (and does so without the risk of losing messages).

Reply:
Loosing debug and trace messages sounds like a good feature to me.

We could remove the neverBlock and stick to the default false. It will still by default drop trace, debug and info. We could change that default to be only trace and debug.

If hitting full queue with those settings then logging is probably used in the wrong way (logging too much).

It's backed by an `ArrayBlockingQueue` so increasing the queueSize seems ok.

Reply:
> the default should be not to lose logging messages.
Agree